app/src/main/java/com/babylon/wallet/android/domain/utils/AccountLockersObserver.kt

```diff
@@ -0,0 +1,134 @@
+                .collect()
+        }
+    }
+
+    fun stopMonitoring() {
+        monitoringJob?.cancel()
+    }
+
+    private suspend fun checkDeposits(
+        authorizedDApps: List<AuthorizedDapp>
+    ): Map<AccountAddress, List<AccountLockerDeposit>> {
+        val dAppsWithAccountLockers = getDAppWithAccountLockers(authorizedDApps)
+        val lockersPerUserAccount = getLockersPerUserAccount(dAppsWithAccountLockers, authorizedDApps)
+
+        return lockersPerUserAccount.mapValues { entry ->
+            val availableDeposits = accountLockersRepository.getAvailableAccountLockerDeposits(
+                accountAddress = entry.key,
+                lockerAddresses = entry.value.mapNotNull { it.lockerAddress }.toSet()
+            ).getOrNull().orEmpty()
+
+            entry.value.filter { it.lockerAddress in availableDeposits }
+                .mapNotNull { dAppWithClaimableItems ->
+                    AccountLockerDeposit(
+                        dAppName = dAppWithClaimableItems.name.orEmpty(),
+                        lockerAddress = dAppWithClaimableItems.lockerAddress ?: return@mapNotNull null
+                    )
+                }
+        }
+    }
+
+    private suspend fun getDAppWithAccountLockers(
+        authorizedDApps: List<AuthorizedDapp>
+    ): List<DApp> {
+        val dAppDefinitionAddresses = authorizedDApps.map { it.dappDefinitionAddress }.toSet()
+        return getDAppsUseCase(dAppDefinitionAddresses, true)
+            .getOrNull()
+            ?.filter { it.lockerAddress != null }
+            .orEmpty()
+    }
+
+    private fun getLockersPerUserAccount(
+        dAppsWithAccountLockers: List<DApp>,
+        authorizedDApps: List<AuthorizedDapp>
+    ): Map<AccountAddress, List<DApp>> {
+        val dAppsWithLockerPerUserAccount = mutableMapOf<AccountAddress, List<DApp>>()
+
+        dAppsWithAccountLockers.forEach { dAppWithAccountLocker ->
+            val authorizedDApp = authorizedDApps.find { it.dappDefinitionAddress == dAppWithAccountLocker.dAppAddress }
+                ?: return@forEach
+
+            authorizedDApp.referencesToAuthorizedPersonas.map { persona -> persona.sharedAccounts?.ids.orEmpty() }
+                .flatten()
+                .forEach { accountAddress ->
+                    val dApps = dAppsWithLockerPerUserAccount[accountAddress] ?: emptySet()
+                    dAppsWithLockerPerUserAccount[accountAddress] = dApps + dAppWithAccountLocker
+                }
+        }
+
+        return dAppsWithLockerPerUserAccount
+    }
+
+    private fun observeAuthorizedDApps(): Flow<List<AuthorizedDapp>> {
+        return dAppConnectionRepository.getAuthorizedDApps()
+            .distinctUntilChanged()
+            .map { dApps ->
+                dApps.filter { dApp ->
+                    dApp.preferences.deposits == AuthorizedDappPreferenceDeposits.VISIBLE
+                }
+            }
+    }
+
+    private fun ticker(): Flow<Unit> {
```

Comment:
not necessary here but better to have it as a val

Reply:
What do you mean?

Reply:
something similar to what i wrote above, but really not so necessary.

Reply:
ah you are right, sorry I missed this